app/notes/[noteId].tsx

```diff
@@ -0,0 +1,86 @@
+import { Link, useFocusEffect, useLocalSearchParams } from 'expo-router';
+import { useCallback, useState } from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { colors } from '../../constants/colors';
+import type { Note as NoteType } from '../../migrations/00003-createTableNotes';
+import type { NoteResponseBodyGet } from '../api/notes/[noteId]+api';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: colors.background,
+    padding: 10,
+  },
+  textContainer: {
+    alignItems: 'center',
+    gap: 12,
+  },
+  title: {
+    fontSize: 24,
+    fontFamily: 'Poppins_700Bold',
+    color: colors.text,
+    textAlign: 'center',
+  },
+  textContent: {
+    fontSize: 16,
+    fontFamily: 'Poppins_400Regular',
+    color: colors.textSecondary,
+    textAlign: 'center',
+    marginTop: 8,
+  },
+});
+
+export default function Note() {
+  const { noteId } = useLocalSearchParams();
+  const [note, setNote] = useState<NoteType | null>(null);
+
+  useFocusEffect(
+    useCallback(() => {
+      async function loadNote() {
+        if (typeof noteId !== 'string') {
+          return;
+        }
+
+        const response = await fetch(`/api/notes/${noteId}`);
+        const responseBody: NoteResponseBodyGet = await response.json();
+
+        if ('note' in responseBody) {
+          setNote(responseBody.note);
+        }
+      }
+
+      loadNote().catch((error) => {
+        console.error(error);
+      });
+    }, [noteId]),
+  );
+
+  if (typeof noteId !== 'string') {
+    return null;
+  }
```

Comment:
@Eprince-hub @ProchaLu This should probably be "not found", right?

Let's do this in a followup PR to `main`

Reply:
With @Eprince-hub, we decided offline this will go to `next`, because it will require an additional database query and a reworking of the "unauthorized" logic

Reply:
@Eprince-hub created 3 PRs:

- https://github.com/upleveled/next-js-example-fall-2024-atvie/pull/11
- https://github.com/upleveled/graphql-example-fall-2024-atvie/pull/2
- https://github.com/upleveled/expo-example-fall-2024-atvie/pull/2